cranelift-wasm/src/translation_utils.rs

```diff
@@ -138,20 +139,38 @@ pub fn tabletype_to_type(ty: wasmparser::Type) -> WasmResult<Option<ir::Type>> {
 }
 
 /// Helper function translating wasmparser block signatures to Cranelift types when possible.
-pub fn blocktype_to_type(ty_or_ft: wasmparser::TypeOrFuncType) -> WasmResult<Option<ir::Type>> {
+pub fn blocktype_to_types<FE: ?Sized + FuncEnvironment>(
+    environ: &mut FE,
+    func: &mut ir::Function,
+    ty_or_ft: wasmparser::TypeOrFuncType,
+) -> WasmResult<Vec<ir::Type>> {
```

Comment:
Maybe use `Cow<'static, [ir::Type]>` to prevent allocation when it is not a multi-value block.

Reply:
Good idea :+1: